packages/askui-nodejs/src/core/cache/cache-entry.ts

```diff
@@ -0,0 +1,51 @@
+import { ControlCommand } from '../ui-control-commands';
+import { CacheEntryReference } from './cache-entry-reference';
+import { logger } from '../../lib';
+
+export class CacheEntry {
+  constructor(
+    public alwaysValid: boolean,
+    public inferenceResponse: ControlCommand,
+    public reference?: CacheEntryReference,
+    public createdAt: Date = new Date(),
+  ) { }
+
+  static fromJson(json: CacheEntry): CacheEntry | undefined {
+    try {
+      if (json === undefined) {
+        throw new Error('Cache entry is undefined');
+      }
+      if (json.inferenceResponse === undefined) {
+        throw new Error('Inference response is required');
+      }
+      if (json.createdAt === undefined) {
+        throw new Error('Created at is required');
+      }
+      if (json.alwaysValid === undefined) {
+        throw new Error('Always valid is required');
```

Comment:
Can you use the property names `json.alwaysValid` instead of "Always valid" in the error message? Please adapt the others too